packages/theme/src/client/modules/info/components/TOC.ts

```diff
@@ -134,13 +135,39 @@ export default defineComponent({
       );
     });
 
+    const updateTocMarker = (): void => {
+      if (!toc.value) return;
+      const tocTop = toc.value.getBoundingClientRect().top;
+      const tocScrollTop = toc.value.scrollTop;
+
+      const activeTocItem = document.querySelector(".toc-item.active");
+
+      if (!activeTocItem) return;
+      const activeTocItemTop = activeTocItem.getBoundingClientRect().top;
+
+      const tocMarker = document.querySelector(".toc-marker") as HTMLElement;
```

Comment:
I think we may add a style property on tocMarker element below, and provide a ref value for it.